includes/options/class-amp-options-manager.php

```diff
@@ -30,7 +30,26 @@ public static function register_settings() {
 			)
 		);
 
-		add_action( 'update_option_' . self::OPTION_NAME, 'flush_rewrite_rules' );
+		add_action( 'update_option_' . self::OPTION_NAME, array( __CLASS__, 'maybe_flush_rewrite_rules' ), 10, 2 );
+	}
+
+	/**
+	 * Flush rewrite rules if the supported_post_types have changed.
+	 *
+	 * @since 0.6.2
+	 *
+	 * @param array $old_options Old options.
+	 * @param array $new_options New options.
+	 */
+	public static function maybe_flush_rewrite_rules( $old_options, $new_options ) {
+		$old_post_types = isset( $old_options['supported_post_types'] ) ? $old_options['supported_post_types'] : array();
+		$new_post_types = isset( $new_options['supported_post_types'] ) ? $new_options['supported_post_types'] : array();
+		sort( $old_post_types );
+		sort( $new_post_types );
+		if ( $old_post_types !== $new_post_types ) {
+			flush_rewrite_rules( false );
+			do_action( 'rri_flush_rules' );
```

Comment:
This call to `do_action( 'rri_flush_rules' )` doesn't exist in the 0.6 branch.  I don't think it's something that should here. Rather if [rewrite-rules-inspector](https://github.com/Automattic/vip-go-mu-plugins/blob/master/rewrite-rules-inspector.php) needs to listen to change to rewrite rule changes, it should instead be adding a `update_option_rewrite_rules` action instead.